tests/kernel/protection/src/main.c

```diff
@@ -48,11 +48,20 @@ void _SysFatalErrorHandler(unsigned int reason, const NANO_ESF *pEsf)
 #define DO_BARRIERS() do { } while (0)
 #endif
 
+#if defined(CONFIG_ARM)
+#define NO_EXECUTE_SUPPORT 1
+#elif defined(CONFIG_X86)
+/* i386 MMU doesn't control execute capabilities, only on x86_64 */
```

Comment:
Do you envision emulating NX support via the CS limit in the future? Or do you not intend to support it at all on ia32 for Zephyr?